### PR TITLE
Frontend: Time tracking page of PTO module

### PR DESCRIPTION
**Pull request by sanjibroy360**
## Notion
[https://www.notion.so/saeloun/Frontend-Time-tracking-page-of-PTO-module-6f30d3ccdbc942d580cf50736d3515bc](https://www.notion.so/saeloun/Frontend-Time-tracking-page-of-PTO-module-6f30d3ccdbc942d580cf50736d3515bc)

## What
- Added the time-off entry form and ensured the user can create, update, delete, and create a duplicate entry.
- Handled minor frontend edge cases.
- Fixed minor bugs in update, delete time-off entry API.

### CODE REVIEW

File: app/javascript/src/components/TimeoffEntries/EntryCard/utils.tsx [hunk 1]
```diff
@@ -0,0 +1,74 @@
+import React from "react";
+
+import { DeleteIcon, EditIcon, CopyIcon } from "miruIcons";
+import { Button, BUTTON_STYLES } from "StyledComponents";
+
+import { Roles } from "constants/index";
+
+export const canEditTimeEntry = (billStatus, role) =>
+  billStatus != "billed" || role == Roles["OWNER"] || role == Roles["ADMIN"];
```
Comment (JijoBose) [inline app/javascript/src/components/TimeoffEntries/EntryCard/utils.tsx:9]: - Let's move the "billed" to  "constants/index"
- There are many billStatus in the app, so let's import it from a constant.
- Please use !== and === for comparing

**Review by JijoBose — CHANGES_REQUESTED** [review]

File: app/javascript/src/components/TimeoffEntries/EntryCard/utils.tsx (hunk 1 above)
Comment (sanjibroy360) [reply app/javascript/src/components/TimeoffEntries/EntryCard/utils.tsx]: Done ✅

**Comment by apoorv1316** [comment]
@sanjibroy360 This is not looking good. We don't have second dropdown so its looking odd.
Screenshot 2024-01-12 at 12 31 12 PM

File: app/javascript/src/components/TimeoffEntries/constants.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
+export const BILL_STATUS = {
+  BILLED: "billed",
+  UNBILLED: "unbilled",
+  NON_BILLABLE: "non_billable",
+};
+
+export const HOLIDAY_TYPES = {
+  NATIONAL: "national",
+  OPTIONAL: "optional",
+};
```
Comment (JijoBose) [inline app/javascript/src/components/TimeoffEntries/constants.ts:10]: Can we move it to "constants/index"

Comment (sanjibroy360) [reply app/javascript/src/components/TimeoffEntries/constants.ts]: Done ✅

File: app/javascript/src/constants/index.tsx [hunk 3]
```diff
@@ -37,6 +37,17 @@ export enum Paths {
   Leave_Management = "/leave-management",
 }
 
+export const BILL_STATUS = {
```
Comment (JijoBose) [inline app/javascript/src/constants/index.tsx:40]: Let's make it an `export enum`

Comment (sanjibroy360) [reply app/javascript/src/constants/index.tsx]: Done ✅

**Comment by sanjibroy360** [comment]
> @sanjibroy360 This is not looking good. We don't have second dropdown so its looking odd. Screenshot 2024-01-12 at 12 31 12 PM

Done ✅ 
Preview:  [https://youtu.be/IwV49S-JnOg](https://youtu.be/IwV49S-JnOg)

cc: @apoorv1316

**Comment by prasanthchaduvula** [comment]
@apoorv1316 Please test the changes now

**Review by apoorv1316 — APPROVED** [review]

**Pull request merged**